src/services/azure-directory.service.ts

```
@@ -295,13 +317,10 @@ export class AzureDirectoryService extends BaseDirectoryService implements IDire
     const memberGroups = await this.client.api(`/users/${user.externalId}/checkMemberGroups`).post({
       groupIds: Array.from(setFilter[1]),
     });
-    if (memberGroups.value.length > 0 && setFilter[0] === UserSetType.IncludeGroup) {
-      return false;
```

Maybe we should throw an error in this case, because this should never occur - and if it does, we probably want to abort syncing as this will invalidate our assumptions in this method.

---

Yeah I removed these thinking they wouldn't be possible conditions anymore, but if I missed anything and it gets to here we should throw an error

---

Reminder to do this if it's called with a group filter. Will be a good safeguard in case it's called from elsewhere in the code, or if someone else misses the refactor.

---

We decided not to do this so all good